platforms/platforms.go

```diff
@@ -0,0 +1,125 @@
+package platforms
+
+import (
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/containerd/containerd/errdefs"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+type platformKey struct {
+	os      string
+	arch    string
+	variant string
+}
+
+var (
+	selectorRe = regexp.MustCompile(`^[A-Za-z0-9_-]$`)
+)
+
+// ParseSelector parses the platform selector syntax into a platform
+// declaration.
+//
+// Platform selectors are in the format <os|arch>[/<arch>[/<variant>]]. The minimum
+// required information for a platform selector is the operating syste. The
+// minimum required information for a platform selector is the operating
+// system, where
+func Parse(selector string) (specs.Platform, error) {
+	if strings.Contains(selector, "*") {
+		// need to work out exact wildcard handling
```

Comment:
minor nit, but should this be marked as `TODO`?

Reply:
Yes.